src/components/dbCalls.js

```diff
@@ -0,0 +1,435 @@
+import { db } from '../utils/firebase';
+import { collection, 
+         getDocs, 
+         query, 
+         where,
+         doc,
+         getDoc,
+         updateDoc,
+         addDoc,
+         orderBy,
+         deleteDoc,
+    } from 'firebase/firestore';
+import UnmapPlayers from './UnmapPlayers';
+
+//fetches all players from database
+const fetchAllPlayersForRoom = async (roomID) => {
+    try {
+        const playerCollectionRef = collection(db, 'rooms', roomID, 'players');
+        const playerSnapshot = await getDocs(playerCollectionRef);
+        return playerSnapshot.docs.map(doc => doc.data().name);
+    }
+    catch (error) {
+        console.error('Error fetching all players: ', error);
+    }
+}
+
+//fetch all alive players from database
+const fetchAlivePlayersForRoom = async (roomID) => {
```

Comment:
This occurs a number of times in this file, but you can combine this with the below method and make "isAlive" a parameter that you pass in. That'll help reduce code duplication